csv_to_geo_yaml

```diff
@@ -94,34 +93,43 @@ addresses.each do |address|
   end
 end
 
-locations.each do |location|
-  next if location[:results].any?
-  # we don't have any lat long data so let's search for it
-  search_string = [location[:address], location[:city], location[:state], location[:zip]].compact.join(", ")
-
-  geo_coder_results = Geocoder.search(search_string)
-
-  geo_coder_results.each do |geo_coder_result|
-    result = {}
-
-    geo_location = geo_coder_result.data['geometry']['location']
-    result[:address] = geo_coder_result.data['formatted_address']
-    result[:location_type] = geo_coder_result.data['geometry']['location_type']
-    result[:latitude] = geo_location['lat']
-    result[:longitude] = geo_location['lng']
-    result[:search_string] = search_string
-
-    location[:results] << result
+start_time = Time.now
+lookups_done = 0
+locations.each_slice(100) do |batch|
+  threads = []
+  batch.each do |location|
+    next if location[:results].any?
```

Comment:
sometimes CS provides us with lat/longs in the csv.  If they do, there are already results and we don’t need to look up any others.